core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -469,10 +469,15 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
       reconfigurable.reconfigure(newConfig)
   }
 
-  private def updatedConfigs(newProps: java.util.Map[String, _], currentProps: java.util.Map[_, _]): mutable.Map[String, _] = {
-    newProps.asScala.filter {
+  private def updatedConfigs(newProps: java.util.Map[String, _], currentProps: java.util.Map[String, _]):
```

Comment:
We should have JavaDoc for this function to identify what it returns.  Also, this is a nitpick, but considering that deletedKeySet is returned as the second part of the tuple, why not put the code for calculating it second in the function?

Reply:
Just pushed a new version that computes `deletedKeySet` second and has a javadoc for the function.